shared/clearwater/sticky_component.rb

```diff
@@ -1,79 +1,48 @@
 require 'clearwater/component'
-require 'clearwater/black_box_node'
 
 module Clearwater
   class StickyComponent
     include Clearwater::Component
 
-    attr_accessor :props, :renderer
-    attr_reader :children
-
-    def self.render props=nil, children=nil
-      unless `props === nil || props.$$is_hash`
-        children = props
-        props = nil
+    class << self
+      alias_method :_new, :new
+      def new(*args)
```

Comment:
I think I land on the side that if you define a `StickyComponent`, then you have a `StickyComponent`. If you need a component to behave both sticky and non-sticky, just make a sticky wrapper component for a non-sticky component. Well defined responsibilities for each type of component will encourage smaller composable classes.